pages/common/git-rebase.md

```diff
@@ -1,23 +1,24 @@
 # git rebase
 
 > Apply local commits on top of another branch's history.
+> This effectively "moves" an entire branch to another base, by creating copies of the commits in the new location.
 
-- Rebase your local branch interactively with the latest changes in local master:
+- Rebase the current branch on top of the master branch:
 
-`git rebase -i master`
+`git rebase master`
 
-- Rebase your local branch interactively with the latest changes from upstream:
+- Start an interactive rebase, which allows the commits to be reordered, omitted, combined or modified:
 
-`git fetch origin; git rebase -i origin/master`
+`git rebase -i {{target_base_branch}}`
 
-- Handle an active rebase merge failure, after editing conflicting file(s):
+- Continue a rebase that was interrupted by a merge failure, after editing conflicting files:
 
 `git rebase --continue`
 
-- Abort a rebase in-progress:
+- Abort a rebase in progress (e.g. if it is interrupted by a merge conflict):
 
 `git rebase --abort`
 
-- Rebase your local branch by specifying new base commit and old base commit:
+- Rebase a branch starting from a specific base commit, rather than the common ancestor shared with the target branch:
 
 `git rebase --onto {{new_base_commit}} {{old_base_commit}}`
```

Comment:
If you ask me, the `--onto` usage is really confusing. Usually I come at it from a command like

    git rebase master

, which (in my head) means "rebase the current branch onto master". What it really means is "rebase the commits `master..HEAD` onto master", but that's not what I'm thinking. The "master" argument is used for **two things**: defining the start of the range of commits to rebase, and defining the attachment point for the rewritten commits.

Then I think to myself "no, I don't want to rebase all of the commits on my current branch, just the last three". What I'm changing relative to the earlier command is the starting point of the range of commits to rebase. It is totally unintuitive to me that I have to type

    git rebase --onto master HEAD^^^

because the `--onto` option makes it look like I'm changing the destination of the rebase. But that is almost never the case; when I resort to `--onto` I'm almost always adjusting the starting point of the range.

But all that whining doesn't change the status quo, which I would express as:

* `git rebase master` means roughly "transplant the commits `master..HEAD` onto branch `master`."
* `git rebase --onto master topicA` means roughly "transplant the commits `topicA..HEAD` onto commit `master`."
* `git rebase --onto master topicA topicB` means roughly "switch to branch `topicB`, then transplant the commits `topicA..topicB` onto commit `master`."

Remember that in all these cases, the notation `A..B` means "all of the commits that are in `B` but are not in `A`, as usual in Git.